### PR TITLE
add Observer client cert auth for Applens backend

### PR DESCRIPTION
**Pull request by yifguoMSFT**

### CODE REVIEW

**Review by SharmaAjay19 — COMMENTED** [review]
We may be able to reuse the GenericCertLoaderService instead of creating a dedicated one.
https://github.com/Azure/Azure-AppServices-Diagnostics-Portal/blob/main/ApplensBackend/Services/GenericCertLoaderService/GenericCertLoader.cs

File: ApplensBackend/Configuration/Startup.cs [hunk 1]
```diff
@@ -58,6 +58,7 @@ public void ConfigureServices(IServiceCollection services)
             services.AddSingleton(Configuration);
 
             GenericCertLoader.Instance.Initialize();
+            SupportObserverCertLoader.Instance.Initialize(Configuration);
```
Comment (SharmaAjay19) [inline ApplensBackend/Configuration/Startup.cs:61]: Quick check - we may need to update the Startup file for NationalCloud as well.

**Comment by yifguoMSFT** [comment]
> We may be able to reuse the GenericCertLoaderService instead of creating a dedicated one. https://github.com/Azure/Azure-AppServices-Diagnostics-Portal/blob/main/ApplensBackend/Services/GenericCertLoaderService/GenericCertLoader.cs

Hi Ajay, I already used the GenericCertLoader in the SupportObserverCertLoader, see [here](https://github.com/Azure/Azure-AppServices-Diagnostics-Portal/blob/4b877d51be82881116bfdeb08dc24726d1e1382b/ApplensBackend/Services/ObserverClientService/SupportObserverCertLoader.cs#L22). It's just for reading the Observer appsetting and decide whether load the cert or not and throw an exception when cert is not found

File: ApplensBackend/Configuration/Startup.cs (hunk 1 above)
Comment (yifguoMSFT) [reply ApplensBackend/Configuration/Startup.cs:61]: Didn't see the configuration files for sovereign cloud in the project, will check with Hawk regarding to this.

And I haven't enabled the client cert yet, so SupportObserverCertLoader here won't load anything which should be fine.

Comment (yifguoMSFT) [reply ApplensBackend/Configuration/Startup.cs:62]: chatted with Hawk, created appsettings.<cloud>.json for different cloud settings and added cert loader to StartupNationalCloud.cs

**Review by SharmaAjay19 — APPROVED** [review]

**Pull request merged**